msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -52,12 +53,26 @@ public abstract partial class ClientApplicationBase
         protected const string DefaultAuthority = "https://login.microsoftonline.com/common/";
 
         /// <summary>
-        /// 
+        /// Constructor of the base application
         /// </summary>
-        /// <param name="clientId"></param>
-        /// <param name="authority"></param>
-        /// <param name="redirectUri"></param>
-        /// <param name="validateAuthority"></param>
+        /// <param name="clientId">Client ID (also named Application ID) of the application as registered in the 
+        /// application registration portal (https://aka.ms/msal-net-register-app)</param>
+        /// <param name="authority">Authority of the Security service token (STS) from which MSAL.NET will acquire the tokens.
+        /// 
+        /// Usual authorities are:
+        /// <list type="bullet">
+        /// <item><c>https://login.microsoftonline.com/tenant/</c>, where <c>tenant</c> is the tenant ID of the Azure AD tenant
+        /// or a domain associated with this Azure AD tenant, in order to sign-in user of a specific organization only</item>
+        /// <item><c>https://login.microsoftonline.com/common/</c> to signing users with any work and school accounts or Microsoft personal account</item>
+        /// <item><c>https://login.microsoftonline.com/organizations/</c> to signing users with any work and school accounts</item>
+        /// <item><c>https://login.microsoftonline.com/consumers/</c> to signing users with only personal Microsoft account (live)</item>
+        /// </list>
+        /// Note that this setting needs to be consistent with what is declared in the application registration portal
+        /// </param>
+        /// <param name="redirectUri">URL where the STS will call back the application with the security token</param>
+        /// <param name="validateAuthority">Boolean telling MSAL.NET if the authority needs to be verified against a list of known authorities. 
+        /// This should be set to <c>false</c> for Azure AD B2C authorities as those are customer specific (a list of known B2C authorities
+        /// cannot be maintained by MSAL.NET</param>
```

Comment:
Feature request: We could however let the developer provide a list on their own... say they want to ensure that for their app only certain authorities was allowed?

Reply:
Yes, I've started thinking of this (and @shoatman as well). I'll propose a design review doc at some point;